sky/clouds/service_catalog/common.py

```diff
@@ -96,23 +96,37 @@ def _need_update() -> bool:
         if _need_update():
             url = f'{constants.HOSTED_CATALOG_DIR_URL}/{constants.CATALOG_SCHEMA_VERSION}/{filename}'  # pylint: disable=line-too-long
             update_frequency_str = ''
+            filename_str = f' ({filename})'
             if pull_frequency_hours is not None:
                 update_frequency_str = f' (every {pull_frequency_hours} hours)'
-            with backend_utils.safe_console_status(
-                    f'Updating {cloud} catalog{update_frequency_str}'):
+            with backend_utils.safe_console_status((f'Updating {cloud} catalog'
+                                                    f'{filename_str}'
+                                                    f'{update_frequency_str}')):
                 try:
                     r = requests.get(url)
                     r.raise_for_status()
                 except requests.exceptions.RequestException as e:
-                    logger.error(f'Failed to download {cloud} catalog:')
-                    with ux_utils.print_exception_no_traceback():
-                        raise e
-            # Save the catalog to a local file.
-            os.makedirs(os.path.dirname(catalog_path), exist_ok=True)
-            with open(catalog_path, 'w') as f:
-                f.write(r.text)
-            with open(meta_path + '.md5', 'w') as f:
-                f.write(hashlib.md5(r.text.encode()).hexdigest())
+                    ux_utils.console_newline()
```

Comment:
For this one here, how about explicitly `close` the `status_console`, since we don't need that status after the warning/error is printed, o.e."
```
with backend_utils.safe_console_status((f'Updating {cloud} catalog'
                                                    f'{filename_str}' as status:
    ...
    status.stop()
    error_str = ...
```

Reply:
That makes sense. I have put it after the `ux_utils.console_newline()` for now:
```
except requests.exceptions.RequestException as e:
    ux_utils.console_newline()
    status.stop()
    error_str = ...
```

This way the spinner message stays in the terminal output:
```
$ sky launch
⠴ Updating AWS catalog: aws/images.csv (every 7 hours)
W 12-16 21:54:50 common.py:114] Failed to fetch AWS catalog aws/images.csv. Using cached catalog files.
I 12-16 21:54:50 optimizer.py:606] == Optimizer ==
...
```

If I put it without the `ux_utils.console_newline()` like so:
```
except requests.exceptions.RequestException as e:
    status.stop()
    error_str = ...
```

The spinner message instantly disappears and only the error is left in the terminal:
```
$ sky launch
W 12-16 21:54:50 common.py:114] Failed to fetch AWS catalog aws/images.csv. Using cached catalog files.
I 12-16 21:54:50 optimizer.py:606] == Optimizer ==
...
```

I'm keen on letting the spinner message stay in the terminal window so its easier for users to know which spinner/step failed. But if you think otherwise lmk and I can change it.

Reply:
That makes sense! Keeping the spinner message looks good to me!